src/test/regress/expected/insert_select_repartition.out

```diff
@@ -761,7 +761,7 @@ WITH r AS (
 )
 INSERT INTO target_table SELECT source_table.a, max(source_table.b) FROM source_table NATURAL JOIN r GROUP BY source_table.a;
 DEBUG:  INSERT target table and the source relation of the SELECT partition column value must be colocated in distributed INSERT ... SELECT
-DEBUG:  only SELECT, UPDATE, or DELETE common table expressions may be router planned
+DEBUG:  router planner does not support queries that reference non-colocated distributed tables
```

Comment:
The result is still the same, but this test is not testing what was intended anymore. I feel like for most of these changed tests we should set `citus.enable_non_colocated_router_query_pushdown` temporarily to `true` so that the same error paths are being hit. Or we should change the queries so that the use colocated shards, but that seems like a lot more work.

Reply:
> The result is still the same, but this test is not testing what was intended anymore. 

This was the case for insert_select_repartition, so I've temporarily set that GUC to ON for this specific INSERT command.

But this is not the case for the others. insert_select_single_shard_table / query_single_shard_table tests were added by me and they were returning a somewhat meaningless debug message before (`found no worker with all shard placements`); and they were aiming to test the bits around non-colocated shards.

And that seems to be the case for the other two --sometimes with a different debug message though-- (`Router planner cannot handle multi-shard select queries`).